### DISCUSSION TITLE
"Cannot connect to the docker socket. Cannot proceed."

### DISCUSSION THREAD

**Discussion by Saiv46** [nextcloud/all-in-one#1411] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install Docker Desktop
2. Run the command from README
```
# For x64 CPUs:
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

### Expected behavior 
Containers should start properly (like [Portainer](https://docs.portainer.io/start/install/server/docker/linux#deployment))

### Actual behavior 
Containers fails with messages:
```
Cannot connect to the docker socket. Cannot proceed.
```

### Host OS 
```env
NAME="Linux Mint"
VERSION="21 (Vanessa)"
ID=linuxmint
ID_LIKE="ubuntu debian"
PRETTY_NAME="Linux Mint 21"
VERSION_ID="21"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=vanessa
UBUNTU_CODENAME=jammy
```

#### Nextcloud AIO version 
`latest`

**Comment by szaimen** [comment]
Can you run `sudo ss -x -a | grep docker.sock`? What does it return?

**Reply by Saiv46** [reply]
It returns
```
u_str LISTEN 0      4096               /home/saiv46/.docker/desktop/docker.sock 2145964            * 0             
u_str ESTAB  0      0                  /home/saiv46/.docker/desktop/docker.sock 2147828            * 2149123       
u_str ESTAB  0      0                  /home/saiv46/.docker/desktop/docker.sock 2145988            * 2148863
```

I also tried to replace `/var/run/docker.sock` with `/home/saiv46/.docker/desktop/docker.sock` and made a symlink

**Reply by havkacik** [reply]
I am having the same issue with Docker Desktop on W10 with WSL2 Ubuntu-22.04. 

`sudo ss -x -a | grep docker.sock` returns: 

`u_str LISTEN 0      4096   /var/run/docker.sock 22186             * 0`


any help needed? I tried many things, spent 3 days  debugging and just can't run nextcloud in docker

**Reply by szaimen** [reply]
Hi, did you follow https://nextcloud.com/blog/your-guide-to-the-nextcloud-all-in-one-on-windows-10-11/?

**Reply by szaimen** [reply]
I have the assuption that the Docker Desktop update v4.19 broke things. See https://github.com/nextcloud/all-in-one/issues/2450

**Reply by havkacik** [reply]
@szaimen, yes I followed the manual you referred to. I suspect new Docker Desktop version 4.19 broke NextCloud functionality, too. 

Desktop Docker **4.16** running on another Windows 11 machine has been working well with default WSL2 (docker-desktop, docker-desktop-data). 

I also installed older version Docker Desktop to **v4.18** in Windows 10 machine and it's working smoothly with default WSL2 (docker-desktop, docker-desktop-data). 

If you can fix this for the new Docker Desktop v4.19, that would be great.

**Reply by szaimen** [reply, marked as answer]
> If you can fix this for the new Docker Desktop v4.19, that would be great.

This needs to be fixed in Docker Desktop unfortunately.

See https://github.com/docker/for-win/issues/13447 and https://github.com/docker/for-mac/issues/6823

**Comment by szaimen** [comment]
All right!

Please try this one then:

```
# For x64 CPUs:
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /home/saiv46/.docker/desktop/docker.sock:/var/run/docker.sock:ro \
-e DOCKER_SOCKET_PATH=/home/saiv46/.docker/desktop/docker.sock \
nextcloud/all-in-one:latest
```

(I've adjusted `--volume /var/run/docker.sock:/var/run/docker.sock:ro` and added `DOCKER_SOCKET_PATH`)

**Reply by Saiv46** [reply]
Same error

**Comment by szaimen** [comment]
all right, then I don*t have many ideas what could be the problem. It works over here without any problem.

I can only suggest to switch to "normal" docker and to not use docker Desktop for this as it is much easier with "normal" docker.

Apart from that I've found this for you: https://docs.docker.com/desktop/faqs/linuxfaqs/#how-do-i-enable-file-sharing

**Reply by Saiv46** [reply]
If it was file sharing problem, then it would output:
```
docker: Error response from daemon: Mounts denied: 
The path /home/saiv46/.docker/desktop/docker.sock is not shared from the host and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> Resources -> File Sharing.
See https://docs.docker.com/ for more info.
```

Instead, something inside Nextcloud container errors on non-root docker

**Reply by szaimen** [reply]
> Instead, something inside Nextcloud container errors on non-root docker

sorry, no idea. As I said, it works over here and I even have tested it successfully with docker rootles.

**Reply by Saiv46** [reply]
Well, thanks for help then.

**Comment by szaimen** [comment]
But I have one idea left.

Could you try this one?

```
# For x64 CPUs:
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /home/saiv46/.docker/desktop/docker.sock.raw:/var/run/docker.sock:ro \
-e DOCKER_SOCKET_PATH=/home/saiv46/.docker/desktop/docker.sock.raw \
nextcloud/all-in-one:latest
```

**Reply by Saiv46** [reply]
No luck

**Comment by szaimen** [comment]
Just wanted to leave a code pointer here but no idea why it fails on you: https://github.com/nextcloud/all-in-one/blob/30911356c1b774c2e651678dd36d27fed1324e70/Containers/mastercontainer/start.sh#L17-L50

**Comment by Weax** [comment]
`--volume /var/run/docker.sock:/var/run/docker.sock:ro \`
try to change /var to //var - this helped me

**Reply by RR-345** [reply]
This "double slash" advice helped me also.  Couple tests below.   

(Success) Two slashes on left side.  One slash on right side: 
`//var/run/docker.sock:/var/run/docker.sock`

(Success) Two slashes on (both) left and right side: 
`//var/run/docker.sock://var/run/docker.sock`

**Reply by Zoey2936** [reply]
the right side should never be changed

**Reply by chuckt62** [reply]
Thanks Weax,
The left side //var fixed it!

**Comment by Seraff** [comment]
Had the same problem. The answer https://github.com/nextcloud/all-in-one/discussions/485 helped.

Installed docker using the recommended method and everything works fine.
```
curl -fsSL get.docker.com | sudo sh
```

**Comment by shalak** [comment]
I have the same issue:
```
nextcloud-aio-mastercontainer  | Docker socket is not available. Cannot continue.
nextcloud-aio-mastercontainer exited with code 0
nextcloud-aio-mastercontainer exited with code 1
nextcloud-aio-mastercontainer exited with code 1
nextcloud-aio-mastercontainer exited with code 1
nextcloud-aio-mastercontainer exited with code 1
nextcloud-aio-mastercontainer exited with code 1
```

Docker was installed using `get.docker.com` script.

The `/var/run/docker.sock` exists:

```
root# ls -al /var/run/docker.sock
srw-rw---- 0 root 29 sty 15:34 /var/run/docker.sock
root# ss -x -a | grep docker.sock
u_str LISTEN 0      4096                           /var/run/docker.sock 133333            * 0
u_str LISTEN 0      4096                               /run/docker.sock 82606             * 0
```

Here is more diagnostics:

```
root# systemctl status docker
● docker.service - Docker Application Container Engine
     Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/docker.service.d
             └─docker.conf, wait-for-cifs.conf, wait-for-nfs.conf
     Active: active (running) since Sun 2023-01-29 15:34:59 CET; 1h 5min ago
TriggeredBy: ● docker.socket
       Docs: https://docs.docker.com
   Main PID: 24743 (dockerd)
      Tasks: 16
     Memory: 848.8M
        CPU: 2min 34.621s
     CGroup: /system.slice/docker.service
             └─24743 /usr/bin/dockerd

root# systemctl status docker.socket
● docker.socket - Docker Socket for the API
     Loaded: loaded (/lib/systemd/system/docker.socket; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-01-27 20:45:17 CET; 1 day 19h ago
   Triggers: ● docker.service
     Listen: /run/docker.sock (Stream)
      Tasks: 0 (limit: 19067)
     Memory: 0B
        CPU: 3ms
     CGroup: /system.slice/docker.socket



root# cat /lib/systemd/system/docker.socket
[Unit]
Description=Docker Socket for the API

[Socket]
ListenStream=/var/run/docker.sock
SocketMode=0660
SocketUser=root
SocketGroup=docker

[Install]
WantedBy=sockets.target
```

I'm starting AIO with following docker-compose:

```yaml
services:
  caddy:
    image: caddy:alpine
    restart: unless-stopped
    container_name: caddy
    volumes:
      - /srv/fast/caddy/Caddyfile:/etc/caddy/Caddyfile
      - /srv/fast/caddy/certs:/certs
      - /srv/fast/caddy/config:/config
      - /srv/fast/caddy/data:/data
      - /srv/fast/caddy/sites:/srv
      - /mnt/ssl:/mnt/ssl
    network_mode: "host"

  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    ports:
      - "8080:8080"
    environment:
      - APACHE_PORT=11000
      - NEXTCLOUD_DATADIR="/srv/fast/ncdata"
      - NEXTCLOUD_MEMORY_LIMIT=2048M
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    depends_on:
      - caddy

volumes:
  nextcloud_aio_mastercontainer:
```

I wonder why do we mount the `/var/run/docker.sock` with `ro`? Doesn't it force the socket to be read-only?

**Reply by shalak** [reply]
Ok, now I see what I did wrong :) I must've removed the `volumes:` key and volumes ended up in `environment:` 🤦  anyhow - my question still stands - why is it supposed to be mounted as `ro`?

**Reply by szaimen** [reply]
ro or rw doesnt make a difference honestly. It will work with both.

**Reply by shalak** [reply]
How is it possible? Docker daemon should not allow AIO to write anything to the socket, thus making it not work at all! What am I missing here?

**Reply by szaimen** [reply]
AIO speaks via http to the docker socket and does not make any changes to it itself. That is why ro works as well.

**Comment by ArmenMadoyan** [comment]
in my MacBook I changed  --volume /var/run/docker.sock:/var/run/docker.sock:ro to
 --volume /var/run/docker.sock.raw:/var/run/docker.sock:ro and it worked fine

**Reply by szaimen** [reply]
Indeed: https://github.com/nextcloud/all-in-one#how-to-run-aio-on-macos

**Comment by z3r0r4** [comment]
I might have found a solution to this!
I was able to get nextcloud aio up and running and controllable from windows and docker desktop by running the command in the wsl2 terminal instead of the powershell or windows cmd.
The command i used was ` sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 80:80 --publish 8080:8080 --publish 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest`
with the output:
`Trying to fix docker.sock permissions internally...   
Creating docker group internally with id 1001
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
Initial startup of Nextcloud All-in-One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080
If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
[02-May-2023 13:15:59] NOTICE: fpm is running, pid 134
[02-May-2023 13:15:59] NOTICE: ready to handle connections
{"level":"info","ts":1683033359.0319157,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}`

**Reply by szaimen** [reply]
Thanks for the workaround! However this is still a bug of Docker Desktop 4.19 and needs to be fixed! See https://github.com/nextcloud/all-in-one/issues/2450

**Reply by z3r0r4** [reply]
Youre welcome! Maybe the answer marker should be switched to this then

**Comment by szaimen** [comment]
Possibly this would have helped: 
https://github.com/nextcloud/all-in-one/discussions/485#discussioncomment-6088261

**Comment by jastranlove2020** [comment]
2025-05-19
Windows Docker 4.41.2
- Container could not run with command prompt due to the error "Cannot connect to the docker socket. Cannot proceed." persisted. Run with WSL it works fine. The GUI can start and stop normal afterward

brian@ServerUPro:~$ sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --pu
blish 8002:80 --publish 8081:8080 --publish 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --vo
lume /mnt/wsl/docker-desktop-bind-mounts/Ubuntu-22.04/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
Trying to fix docker.sock permissions internally...
Creating docker group internally with id 1001
WARNING: No blkio throttle.read_bps_device support

**Comment by wijnen** [comment]
Since the update to the nextcloud-aio container from a few days ago, I now have the same error on my Synology NAS and I have found the problem in my case. To figure it out, I read /start.sh and saw that it runs "docker info" to see if the socket works. If it doesn't, it gives that error.

To check what happened, I tried the docker run command with the extra options `--tty --interactive --entrypoint /bin/bash`.  From there, I attempted to run `docker info`. That resulted in this output:
```
0ba7e1490e9a:~/docker-aio# docker info
Client:
 Version:    29.0.2
 Context:    default
 Debug Mode: false

Server:
ERROR: Error response from daemon: client version 1.52 is too new. Maximum supported API version is 1.43
errors pretty printing info
0ba7e1490e9a:~/docker-aio#
```

So obviously, docker on my NAS is too old. But it doesn't look like I can upgrade it. So my question is now: how can I get a version of nextcloud aio that uses the older docker?

**Reply by szaimen** [reply]
Hi, see https://github.com/nextcloud/all-in-one/issues/7223

**Reply by wijnen** [reply]
Thanks!

For anyone finding this post, who does not want to read through that thread: the solution (really workaround) is to add `--env DOCKER_API_VERSION=1.43` to the command line of `docker run`.

It works fine for me.
